### PR TITLE
Provide `python` symlink for Python3

### PR DESCRIPTION
**Pull request by aneeshusa**
###### Motivation for this change

Part of #18185.
cc @FRidh @domenkozar 
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/python/cpython/3.5/default.nix [hunk 1]
```diff
@@ -89,6 +89,9 @@ in stdenv.mkDerivation {
 
     # Python on Nix is not manylinux1 compatible. https://github.com/NixOS/nixpkgs/issues/18484
     echo "manylinux1_compatible=False" >> $out/lib/${libPrefix}/_manylinux.py
+
+    # Use Python3 as default python
+    ln -s "$out/bin/python3" "$out/bin/python"
```
Comment (NeQuissimus) [inline pkgs/development/interpreters/python/cpython/3.5/default.nix]: What happens if I also install python 2? Will that not be confusing?

Comment (aneeshusa) [inline pkgs/development/interpreters/python/cpython/3.5/default.nix]: The `python-wip` branch is meant to sniff out breakage caused by Python3 as default on Hydra, so I've added this to ensure `patchShebangs` works. In the Python3-as-default future (where the default install doesn't include Python2), we'll need this to provide a `python` executable. As I mentioned in the issue, we should also stop `python2` from installing a `python` executable to allow simultaneous installations, but IMO that can wait as well (shouldn't affect Hydra results).

Confusion with any change like this is inevitable; I think the best way to address it is to wait until next release to do the actual switchover, and to clearly mark this as a breaking change in the release notes.

**Comment by domenkozar** [comment]
This used to be a problem since python3 was higher priority, but may work
now that it's default.

**Comment by aneeshusa** [comment]
@domenkozar I'm not seeing a jobset for `python-wip` at https://hydra.nixos.org/project/nixpkgs, can you start one? Best way to find out is to try it :)

**Comment by FRidh** [comment]
@aneeshusa `python-wip` is occasionally used to test changes. It's tracked by another Hydra, https://headcounter.org/hydra/jobset/iElectric/python-wip

I will create a new branch here called `python3` where we can work on. Whenever we want to build against this Hydra, I can push it to `python-wip`. If we get a bit further we can also ask for a job on `hydra.nixos.org`

**Comment by FRidh** [comment]
@aneeshusa I've created the `python3` branch. Please base any further changes regarding the migration on this branch. I've also included these commits, and they're also on the `python-wip` branch currently. I'll keep this PR open until it landed in staging or master.

File: pkgs/development/interpreters/python/cpython/3.5/default.nix (hunk 1 above)
Comment (FRidh) [inline pkgs/development/interpreters/python/cpython/3.5/default.nix]: I've made this change for all interpreter versions and also all binaries in /bin
164fd16749d6ca123d54efbee739a68dc4e06344

**Comment by aneeshusa** [comment]
Closing in favor of #19626, which includes these changes.

**Pull request closed without merging**